lgc/state/Compiler.cpp

```diff
@@ -145,31 +145,26 @@ void PipelineState::generate(std::unique_ptr<Module> pipelineModule, raw_pwrite_
   Timer *codeGenTimer = timers.size() >= 3 ? timers[2] : nullptr;
 
   // Set up "whole pipeline" passes, where we have a single module representing the whole pipeline.
-  //
-  // TODO: The "whole pipeline" passes are supposed to include code generation passes. However, there is a CTS issue.
-  // In the case "dEQP-VK.spirv_assembly.instruction.graphics.16bit_storage.struct_mixed_types.uniform_geom", GS gets
-  // unrolled to such a size that backend compilation takes too long. Thus, we put code generation in its own pass
-  // manager.
-  std::unique_ptr<PassManager> patchPassMgr(PassManager::Create());
-  patchPassMgr->setPassIndex(&passIndex);
-  patchPassMgr->add(createTargetTransformInfoWrapperPass(getLgcContext()->getTargetMachine()->getTargetIRAnalysis()));
+  std::unique_ptr<PassManager> passMgr(PassManager::Create());
+  passMgr->setPassIndex(&passIndex);
+  passMgr->add(createTargetTransformInfoWrapperPass(getLgcContext()->getTargetMachine()->getTargetIRAnalysis()));
```

Comment:
Do we still need this? Doesn't addTargetPasses in the back-end add it for us?

Reply:
If I remove this line, llpc lit tests start failing:
    LLPC_SHADERTEST :: shaderdb/ObjInput_TestIndexingInterpOfInputArray_lit.frag
I will investigate why.

Reply:
No the backend does not add it. I think it is usually up to the driver to do this -- at least "opt" does it before running IR passes that might depend on codegen-specific information. Perhaps this line should move to Patch::addOptimizationPasses, or somewhere like that?

Reply:
OK. I think it is fine to leave it here then.

Reply:
Or perhaps it could move into LgcContext::preparePassManager along with the TargetLibraryInfoWrapperPass creation?